Sources/Gtk/Widgets/Widget.swift

```diff
@@ -237,41 +241,10 @@ open class Widget {
         }
     }
 
-    public var topMargin: Int {
-        get {
-            return Int(gtk_widget_get_margin_top(widgetPointer))
-        }
-        set {
-            gtk_widget_set_margin_top(widgetPointer, gint(newValue))
-        }
-    }
-
-    public var bottomMargin: Int {
-        get {
-            return Int(gtk_widget_get_margin_bottom(widgetPointer))
-        }
-        set {
-            gtk_widget_set_margin_bottom(widgetPointer, gint(newValue))
-        }
-    }
-
-    public var leftMargin: Int {
-        get {
-            return Int(gtk_widget_get_margin_start(widgetPointer))
-        }
-        set {
-            gtk_widget_set_margin_start(widgetPointer, gint(newValue))
-        }
-    }
-
-    public var rightMargin: Int {
-        get {
-            return Int(gtk_widget_get_margin_end(widgetPointer))
-        }
-        set {
-            gtk_widget_set_margin_end(widgetPointer, gint(newValue))
-        }
-    }
+    @GObjectProperty(named: "margin-top") public var topMargin: Int
+    @GObjectProperty(named: "margin-bottom") public var bottomMargin: Int
+    @GObjectProperty(named: "margin-start") public var leadingMargin: Int
+    @GObjectProperty(named: "margin-end") public var trailingMargin: Int
 
     public var horizontalAlignment: Align {
```

Comment:
Is it possible to update these properties too with GValueRepresentableEnum?

Reply:
Yep, I've already used it in the constraints branch so I know it works. Didn't know if you would be okay with me just changing all the properties.